Assets/Scripts/Room.cs

```diff
@@ -19,9 +19,30 @@ public class Room : MonoBehaviour
 
     public UnityEvent<string> roomEnteredEvent = new UnityEvent<string>();
 
+    public string roomtype = "normal";
```

Comment:
suggest using an enum or even a bool over a string here, or better yet split up the behavior that differentiates "randomized enemies room", "boss room", and "enemy-free/tutorial room" into separate classes

Reply:
It will be left to later work.

Reply:
Just wanted to point out that your branch uncovered a bug that could have been mitigated by such an effort.  Due to reliance on run-time values "openWhenEnemiesCleared" and now "roomtype" to determine behavior instead of explicit classes/types, you can have a "Room" which is "normal", so it spawns random enemies, but which is not also marked as "openWhenEnemiesCleared".  This will cause these rooms, such as Room 5 and Room 7, to bypass Destroy()ed enemy "cleanup" in Update(), which will yield a null pointer exception when re-entering and "Activating" dead enemies.  If there were, for instance, explicit "EnemyRoom" and "EnemyFreeRoom" classes (with no enemies) there wouldn't be an opportunity to produce an invalid state (i.e. "normal" + "openWhenEnemiesCleared=false") since an enemy-free room wouldn't have a list of potentially-destroyed "enemies" to activate when it is entered, and an enemy room would actively remove any Destroy()ed enemies on every update.